### PR TITLE
Migrate keyboard-navigable-blocks e2e tests from puppeteer to playwright

### PR DESCRIPTION
**Pull request by jeryj**
## What?
Migrating keyboard-navigable-blocks e2e tests from puppeteer to playwright

## Why?

Part of the larger migration effort in https://github.com/WordPress/gutenberg/issues/38851, also because I'm going to be modifying this test as part of https://github.com/WordPress/gutenberg/pull/54513 and figured I may as well migrate it before making changes.

## How?

Followed the [migration guide](https://github.com/WordPress/gutenberg/blob/8e7ce7328f67935dc38e09c48004645f93e13cdb/docs/contributors/code/e2e/migration.md).

## Testing Instructions
`npm run test:e2e:playwright -- editor/various/keyboard-navigable-blocks.spec.js`

### CODE REVIEW

**Comment by scruffian** [comment]
The tests pass and the code makes sense to me but I'll let someone with more experience approve.

File: test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+				activeElement.innerText
+			);
+		} );
+
+		expect( ariaLabel ).toBe( label );
+	}
+
+	async navigateToContentEditorTop() {
+		// Use 'Ctrl+`' to return to the top of the editor.
+		await this.pageUtils.pressKeys( 'ctrl+`' );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js]:
```suggestion
		await this.pageUtils.pressKeys( 'ctrl+`',  { times: 5 } );
```

We can use `times` option here and remove duplication.

**Comment by Mamaduka** [comment]
Thank you, @jeryj!

Currently, the test produced ESLint warnings for [`expect-expect`](https://github.com/playwright-community/eslint-plugin-playwright/blob/main/docs/rules/expect-expect.md), as it's recommended to make assertions in the test body.

We might want to refactor the test a bit to satisfy this rule or ignore it at the file level.

P.S. If you have an ESLint add-on, then it makes really annoying viewing the test file 😅

## How to test
```
npm run lint:js test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js
```

## Screenshot
CleanShot 2023-10-02 at 11 58 43

File: test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+		await this.tabThroughBlockToolbar();
+
+		await this.page.keyboard.press( 'Tab' );
+		await this.expectLabelToHaveFocus( 'Block: Paragraph' );
+
+		const blockText = await this.page.evaluate( () => {
+			const { activeElement } =
+				document.activeElement.contentDocument ?? document;
+			return activeElement.innerHTML;
+		} );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js]: Can we use a locator for this?

```js
await expect( blockLocator ).toHaveText( paragraphText );
```

File: test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+		const ariaLabel = await this.page.evaluate( () => {
+			const { activeElement } =
+				document.activeElement.contentDocument ?? document;
+			return (
+				activeElement.getAttribute( 'aria-label' ) ||
+				activeElement.innerText
+			);
+		} );
+
+		expect( ariaLabel ).toBe( label );
```
Comment (Mamaduka) [inline test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js:217]: Btw, the `page.locator( ':focus' )` or `editor.canvas.locator( ':focus' )` can be used for locating active elements.

Comment (jeryj) [reply test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js:217]: I tried refactoring this to use `:focus` but couldn't get the tests to pass. The conditional of checking either the aria label or the innerText was getting in the way. If there was a `toHaveAccessibleName()` check, that would be really useful, but I couldn't find a good way of doing that.

File: test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js (hunk 2 above)
Comment (jeryj) [reply test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js]: `const activeElement = this.editor.canvas.locator( ':focus' );` works well to replace this `blockText` section.

**Comment by jeryj** [comment]
Thank you for the review, @Mamaduka! I've addressed the feedback as best I could.

File: test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js (hunk 3 above)
Comment (Mamaduka) [reply test/e2e/specs/editor/various/keyboard-navigable-blocks.spec.js:217]: Thank you, @jeryj. It makes sense.

**Review by Mamaduka — APPROVED** [review]
Thanks for the followups, @jeryj!

**Pull request merged**